frame/lending/rpc/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use composable_support::rpc_helpers::{SafeRpcWrapper, SafeRpcWrapperType};
+use frame_support::{pallet_prelude::MaybeSerializeDeserialize, Parameter};
+use jsonrpc_core::{Error as RpcError, ErrorCode, Result as RpcResult};
+use jsonrpc_derive::rpc;
+use lending_runtime_api::LendingRuntimeApi;
+use sp_api::ProvideRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::{generic::BlockId, traits::Block as BlockT};
+use sp_std::{marker::PhantomData, sync::Arc};
+
+#[rpc]
+pub trait LendingApi<BlockHash, AccountId, Balance, MarketId>
+where
+	Balance: SafeRpcWrapperType,
+{
+	#[rpc(name = "lending_getBorrowLimit")]
+	fn get_borrow_limit(
+		&self,
+		market_id: MarketId,
+		account: AccountId,
+		at: Option<BlockHash>,
+	) -> RpcResult<SafeRpcWrapper<Balance>>;
+}
+
+pub struct Lending<C, Block> {
+	client: Arc<C>,
+	_marker: PhantomData<Block>,
+}
+
+impl<C, M> Lending<C, M> {
+	pub fn new(client: Arc<C>) -> Self {
+		Self { client, _marker: Default::default() }
+	}
+}
+
+impl<C, Block, AccountId, Balance, MarketId>
+	LendingApi<<Block as BlockT>::Hash, AccountId, Balance, MarketId>
+	for Lending<C, (Block, AccountId, Balance, MarketId)>
+where
+	Block: BlockT,
+	AccountId: Send + Sync + Parameter + MaybeSerializeDeserialize + Ord + 'static,
+	MarketId: Send + Sync + Parameter + MaybeSerializeDeserialize + Ord + 'static,
+	Balance: Send + Sync + 'static + SafeRpcWrapperType,
+	C: Send + Sync + ProvideRuntimeApi<Block> + HeaderBackend<Block> + 'static,
+	C::Api: LendingRuntimeApi<Block, AccountId, Balance, MarketId>,
+{
+	fn get_borrow_limit(
+		&self,
+		market_id: MarketId,
+		remote_account: AccountId,
+		at: Option<<Block as BlockT>::Hash>,
+	) -> RpcResult<SafeRpcWrapper<Balance>> {
+		let api = self.client.runtime_api();
+		let at = BlockId::hash(at.unwrap_or_else(|| {
+			// If the block hash is not supplied assume the best block.
+			self.client.info().best_hash
+		}));
+
+		let runtime_api_result = api.get_borrow_limit(&at, market_id, remote_account);
+		// TODO(benluelo): Review what error message & code to use
+		runtime_api_result.map_err(|e| {
+			RpcError {
+				code: ErrorCode::ServerError(9876), // No real reason for this value
+				message: "Something wrong".into(),
+				data: Some(format!("{:?}", e).into()),
+			}
```

Comment:
Can we sort this out? @benluelo do we have a way of handling his generically?

Reply:
It really depends on what the front end people want, the code and message don't particularly matter since the error is all in `data`

Reply:
This is what `e` is: https://github.com/paritytech/substrate/blob/master/primitives/api/src/lib.rs#L437-L459

Reply:
Alright, then we can perhaps extract this function to `composable-support`?

Reply:
good idea, I can do that after this PR (unless you want to include it in this one)